### PR TITLE
Feature/dvlib-150 pain.008

### PR DESCRIPTION
**Pull request by Gragx**

### CODE REVIEW

File: iso20022-ch-impl/src/main/java/ch/dvbern/oss/lib/iso20022/Iso20022JaxbUtil.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		String schemaLocation) {
+		QName name = new QName(schemaLocation, documentClass.getSimpleName());
+
+		return new JAXBElement<>(name, documentClass, elemToMarshall);
+	}
+
+	private static class PainValidationEventHandler implements ValidationEventHandler {
+		@Override
+		public boolean handleEvent(ValidationEvent event) {
+			throw new Iso20022RuntimeException("Unexpected error while generating pain001 file: "
```
Comment (xfh) [inline iso20022-ch-impl/src/main/java/ch/dvbern/oss/lib/iso20022/Iso20022JaxbUtil.java]: The error message does not match any more "pain001 file" should be replace with something more generic, like "XML file"
Likewise, PainValidationEventHandler is no longer appropriate. How about "JaxbValidationEventHandler"?

File: iso20022-ch-impl/pom.xml [hunk 2]
```diff
@@ -58,12 +58,22 @@
             <artifactId>annotations</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+        </dependency>
```
Comment (xfh) [inline iso20022-ch-impl/pom.xml:63]: Hibernate-validator should not be a required dependency. Instead javax.validation:validation-api should be added as an optional dependency `<optional>true</optional>` (other implementations of the validation-api should be compatible).
hibernate-validator can be changed to `<scope>test</scope>` if required.

File: iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ * Tests for the {@link Pain008Service}.<br>
+ *
+ * The resulting XML will be saved to STORE_PATH if WRITE_TO_FILE is set to true.
+ */
+public class Pain008ServiceTest {
+
+	private static final boolean WRITE_TO_FILE = true;
+	private static final String STORE_PATH = "target/pain008Reference.xml";
+
+	private Pain008V00102CH03Service pain008Service = new Pain008V00102CH03Service();
```
Comment (xfh) [inline iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java]: Field 'pain008Service' may be 'final'

File: iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	private Pain008DTO painDto;
```
Comment (xfh) [inline iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java]: What is the purpose of this field? Why not simply refactor the init() method to return a Pain008DTO?

**Review by xfh — CHANGES_REQUESTED** [review]

File: iso20022-ch-impl/src/main/java/ch/dvbern/oss/lib/iso20022/Iso20022JaxbUtil.java (hunk 1 above)
Comment (Gragx) [reply iso20022-ch-impl/src/main/java/ch/dvbern/oss/lib/iso20022/Iso20022JaxbUtil.java]: Done
Changed both error messages in this service to something more generic.

File: iso20022-ch-impl/pom.xml (hunk 2 above)
Comment (Gragx) [reply iso20022-ch-impl/pom.xml:63]: The tests do not currently cover the bean validation.
As such, replaced hibernate-validator with validation-api

File: iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java (hunk 3 above)
Comment (Gragx) [reply iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java]: done

File: iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java (hunk 4 above)
Comment (Gragx) [reply iso20022-ch-impl/src/test/java/ch/dvbern/oss/lib/iso20022/pain008/v00102ch03/Pain008ServiceTest.java]: done

File: pom.xml [hunk 5]
```diff
@@ -95,9 +95,9 @@
                 <scope>provided</scope>
             </dependency>
             <dependency>
-                <groupId>org.hibernate.validator</groupId>
-                <artifactId>hibernate-validator</artifactId>
-                <version>${version.hibernate}</version>
+                <groupId>javax.validation</groupId>
+                <artifactId>validation-api</artifactId>
+                <version>2.0.1.FINAL</version>
```
Comment (xfh) [inline pom.xml]: Use Version 1.0.1.Final, which is JavaEE7, since we still require EE7

**Review by xfh — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 5 above)
Comment (Gragx) [reply pom.xml]: I'm guessing we want version 1.1.0.Final? 😊

Comment (xfh) [reply pom.xml]: Correct, sorry!

**Review by xfh — APPROVED** [review]

**Pull request merged**